runtime/lua/vim/secure.lua

```diff
@@ -80,34 +80,27 @@ function M.read(path)
   end
 
   -- File either does not exist in trust database or the hash does not match
-  local choice = vim.fn.confirm(
+  local ok, result = pcall(
+    vim.fn.confirm,
     string.format('%s is not trusted.', fullpath),
     '&ignore\n&view\n&deny\n&allow',
     1
   )
 
-  if choice == 0 or choice == 1 then
+  if not ok and result ~= 'Keyboard interrupt' then
+    error(result)
+  elseif not ok or result == 0 or result == 1 then
     -- Cancelled or ignored
     return nil
-  elseif choice == 2 then
+  elseif result == 2 then
     -- View
-    vim.cmd('new')
-    local buf = vim.api.nvim_get_current_buf()
-    local lines = vim.split(string.gsub(contents, '\n$', ''), '\n')
-    vim.api.nvim_buf_set_lines(buf, 0, -1, false, lines)
-    vim.bo[buf].bufhidden = 'hide'
-    vim.bo[buf].buftype = 'nofile'
-    vim.bo[buf].swapfile = false
-    vim.bo[buf].modeline = false
-    vim.bo[buf].buflisted = false
-    vim.bo[buf].readonly = true
-    vim.bo[buf].modifiable = false
+    vim.cmd('sview ' .. fullpath)
```

Comment:
We cannot re-read the file here, as it may have changed between first reading into memory and here. That is the whole reason we went through the whole dance of creating a new buffer and using `buf_set_lines` in the first place.

Reply:
It's important to know that we're evaluating the right thing, but I don't think it particularly matters for simply showing the file. Even if the contents were outdated, next time the file is checked it will be on the new version

Reply:
The point of showing the file is so that the user can inspect it and make a decision or not. If the file contents on disk change somehow then the user is making a decision on something different than they are actually seeing. This seems like a bad user experience to me, and I don't see what the upside is, other than saving a few lines of code.

Reply:
At the point of opening the buffer the calculated hash is discarded. With `:trust` command user always makes the decision based on what he actually sees in the buffer, as that is used to calculate the hash again. The only downside I can think of is that if the file did change and the warning was triggered based on that, the user won't know what it actually was. But that should never happen, in my opinion just making sure we have that handled and never let it through is good enough.